src/mui/input/EmbeddedArrayInput.js

```diff
@@ -0,0 +1,161 @@
+import FlatButton from 'material-ui/FlatButton';
+import React, { Component } from 'react';
+import { FieldArray, Field } from 'redux-form';
+import PropTypes from 'prop-types';
+
+import ContentCreateIcon from 'material-ui/svg-icons/content/create';
+import ActionDeleteIcon from 'material-ui/svg-icons/action/delete';
+
+/**
+ * An Input component for generating/editing an embedded array
+ *
+ *
+ * Use it with any set of input componentents as children, like `<TextInput>`,
+ * `<SelectInput>`, `<RadioButtonGroupInput>` ... etc.
+ *
+ * @example
+ * export const CommentEdit = (props) => (
+ *     <Edit {...props}>
+ *         <SimpleForm>
+ *              <EmbeddedArrayInput source="links">
+ *                  <SimpleList>
+ *                       <TextField source="url" />
+ *                       <TextField source="context"/>
+ *                   </SimpleList>
+ *               </EmbeddedArrayInput>
+ *         </SimpleForm>
+ *     </Edit>
+ * );
+ */
+export class EmbeddedArrayInput extends Component {
+    constructor(props) {
+        super(props);
+        // nothing to do for now
+    }
+
+    renderList = ({ fields }) => {
+        const { children, arrayElStyle } = this.props;
+        return (
+            <div className="EmbeddedArrayInputContainer" style={{ margin: '1em' }}>
+                <div>
+                    {
+                        fields.map((member, index) =>
+                            <ArrayElement key={index} fields={fields} inputs={children} elStyle={arrayElStyle} member={member} index={index} />
+                        )
+                    }
+                </div>
+                <FlatButton primary style={{ float: 'right' }} icon={<ContentCreateIcon />} label="Add" onClick={() => fields.push({})} />
+                <div style={{ clear: 'both' }} />
+            </div>
+        )
+    }
+
+
+    render() {
+        const { input, resource, label, source, allowEmpty, basePath, onChange, children, meta } = this.props;
+
+        return (
+            <FieldArray name={source} component={this.renderList} />
+        )
+    }
+}
+
+const ArrayElement = ({ fields, inputs, member, index, elStyle }) => {
```

Comment:
I've extracted the component to an external file, then i found a similar example in `SelectInput` where the author used a function to render the sub element. So i followed that example since this element does not represent a general component.

If you still prefer to put it as separate component please let me know.